### ISSUE TITLE
Macro names converted to lowercase

### ISSUE THREAD

**Issue by andrew-davie** [dasm-assembler/dasm#10]
Macros have some erroneous behaviour...

    MAC test
    .byte 0
    ENDM

    MAC Test
    .byte 1
    ENDM

    MAC TEST
    .byte 2
    ENDM

    TEST

The above example will generate a '.byte 0' - and ALL of the macro names in the output listing are converted to lowercase - but the use (at the bottom) is left as-is. I think this is a potential issue, as labels elsewhere in the code are case-sensitive.

The issue:  macro names are converted to lowercase. One wonders why.
The second issue: re-use of a macro name with different case  is not reported, 
The third issue: only the first use of a (case insensitive) macro name is actually used.

**Comment by andrew-davie** [comment]
I just noticed this in the documentation...

"V2.12
	-Fixed macro naming bug (macros wouldn't work if the name after
	 the 'mac' was in upper case)."

**Comment by msaarna** [comment]
I took a look at this one, and can offer some answers...

issue the first: macro names are converted to lower-case because they get thrown into the same mnemonic list as opcodes and pseudo-ops, and inherit the same case-insensitivity as opcodes and pseudo-ops. Macros don't show up in the symbol list and aren't handled as such.

issues two and three: dasm even ignores duplicate definitions of a macro, even when it has duplicate case. My best guess is someone thought this was desirable if multiple definitions came in via redundant includes. I think it's misguided since there's no way of knowing if the macro definitions differ, so I added an [error on duplicate macro](https://github.com/dasm-assembler/dasm/commit/fdb5ef8b3c229cc864b41ad9a11d04c01c8ed3f7)

I'm going to close this out, since case-insensitive macros is "as-designed". If you want to open a feature request for case sensitivity, have at it, but heads up that it's not a trivial change.
